notebooks/OrthoIV and DRIV Examples.ipynb

```diff
@@ -0,0 +1,1642 @@
+       "--------------------------------------------------------------------\n",
+       "cate_intercept          10.03   0.14 71.638    0.0    9.755   10.304\n",
+       "--------------------------------------------------------------------\n",
+       "\n",
+       "<sub>A linear parametric conditional average treatment effect (CATE) model was fitted:\n",
+       "$Y = \\Theta(X)\\cdot T + g(X, W) + \\epsilon$\n",
+       "where for every outcome $i$ and treatment $j$ the CATE $\\Theta_{ij}(X)$ has the form:\n",
+       "$\\Theta_{ij}(X) = \\phi(X)' coef_{ij} + cate\\_intercept_{ij}$\n",
+       "where $\\phi(X)$ is the output of the `featurizer` or $X$ if `featurizer`=None. Coefficient Results table portrays the $coef_{ij}$ parameter vector for each outcome $i$ and treatment $j$. Intercept Results table portrays the $cate\\_intercept_{ij}$ parameter.</sub>\n",
+       "\"\"\""
+      ]
+     },
+     "execution_count": 10,
+     "metadata": {},
+     "output_type": "execute_result"
+    },
+    {
+     "data": {
+      "application/javascript": [
+       "\n",
+       "            setTimeout(function() {\n",
+       "                var nbb_cell_id = 10;\n",
+       "                var nbb_formatted_code = \"from econml.iv.dr import LinearIntentToTreatDRIV\\n\\nest5 = LinearIntentToTreatDRIV(model_t_xwz=model_clf())\\nest5.fit(y, T, Z=Z, X=None, W=X)\\nprint(\\\"True Treatment Effect: \\\", func(X))\\nest5.summary()\";\n",
+       "                var nbb_cells = Jupyter.notebook.get_cells();\n",
+       "                for (var i = 0; i < nbb_cells.length; ++i) {\n",
+       "                    if (nbb_cells[i].input_prompt_number == nbb_cell_id) {\n",
+       "                        nbb_cells[i].set_text(nbb_formatted_code);\n",
+       "                        break;\n",
+       "                    }\n",
+       "                }\n",
+       "            }, 500);\n",
+       "            "
+      ],
+      "text/plain": [
+       "<IPython.core.display.Javascript object>"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "from econml.iv.dr import LinearIntentToTreatDRIV\n",
+    "\n",
+    "est5 = LinearIntentToTreatDRIV(model_t_xwz=model_clf())\n",
+    "est5.fit(y, T, Z=Z, X=None, W=X)\n",
+    "print(\"True Treatment Effect: \", func(X))\n",
+    "est5.summary()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Example Usage with Heterogeneous Treatment Effects"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## DGP\n",
+    "In this section, we keep the same data generation process, but only change the treatment effect $\\theta(X)$ to a function of $X$, and learn the heterogeneous treatment effect. We'd like to see the effect interval learnt from each model could recover the true estimate and the coefficient estimates in the final linear model could recover the true coefficient.\n",
```

Comment:
This is great.  However, it looks like X only has one column right now because of how you partition the features into X and W, so X[0] is really all of X; might be good to add a few other columns (which have no effect on the treatment effect) into X instead of W.

Reply:
Yeah I feel there is no big difference, the point is we show the relationship between X[0] and HTE. Given this is pre-defined DGP, we already know HTE only relates to X[0], so I only put X[0] as X.

Reply:
I think that the use case where a user has a set of features and only some of them actually matter might be nice to highlight, but maybe it's not too important and what you've got is okay.